core/lib/dal/src/blocks_dal.rs

```diff
@@ -1395,6 +1396,32 @@ impl BlocksDal<'_, '_> {
         .await?;
         Ok(())
     }
+
+    pub async fn get_fee_address_for_l1_batch(
+        &mut self,
+        l1_batch_number: &L1BatchNumber,
```

Comment:
nit: I think receiving number by reference doesn't make sense there, it's cheaper to copy number than reference it